app/models.py

```diff
@@ -88,10 +91,16 @@ def student_count(cls):
 
     def as_dict(self):
         main = {c.key: getattr(self, c.key) for c in inspect(self).mapper.column_attrs}
-        students = [student.id for student in self.students]
-        stops = [stop.id for stop in self.stops]
-        main['students'] = students
-        main['stops'] = stops
+        main['students'] = [student.as_dict() for student in self.students]
+        main['stops'] = [stop.as_dict() for stop in self.stops]
+        main['school'] = {c.key: getattr(self.school, c.key) for c in inspect(self.school).mapper.column_attrs}
+        main['school']['arrival_time'] = self.school.arrival_time.isoformat()
+        main['school']['departure_time'] = self.school.departure_time.isoformat()
+        main['complete'] = True
+        for student in main['students']:
+            if not student['in_range']:
+                main['complete'] = False
+                break
```

Comment:
Can you not just call as_dict() on school instead of essentially creating the school dict again inside of the route dict?